src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -2229,7 +2229,10 @@ private void ValidateStatementRange(StatementSyntax firstStatement, StatementSyn
                 throw new ArgumentException("statements not within tree");
             }
 
-            if (firstStatement.Parent == null || firstStatement.Parent != lastStatement.Parent)
+            if ((firstStatement.Parent == null || firstStatement.Parent != lastStatement.Parent) &&
```

Comment:
Was this supposed to be `||` instead of `&&`?

Reply:
Nope, the subsquent checks have `not` in them